hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1352,9 +1356,11 @@ private void deleteInternal(Container container, boolean force)
       // empty as a defensive check.
       LOG.error("Could not determine if the container {} is empty",
           container.getContainerData().getContainerID(), e);
-      throw new StorageContainerException("Could not determine if container "
-          + container.getContainerData().getContainerID() +
-          " is empty", DELETE_ON_NON_EMPTY_CONTAINER);
+      String errorMsg =
+          "Could not determine if the container " + container.getContainerData()
+              .getContainerID() + "is empty";
+      triggerVolumeScanAndThrowException(container, errorMsg,
```

Comment:
What is difference here in overall flow, if we detect here Volume faulty on scan or not?  if volume is faulty will container will be removed  and DN will report success?

Reply:
If a volume is found faulty in the scan , it is marked as failed and then the failedVolumeListener will pick it up and remove containers that are present in that volume from the containerSet in memory and send an FCR . SCM will identify these as missing containers and trigger replication elsewhere .

Reply:
okay  so a single container delete failure here will result in volume scan. But if volume is found faulty, all containers within that will be removed and then FCR will be sent.

Reply:
Yes right , that is how it should behave as containers in faulty volumes cannot be accessed.